### PR TITLE
JDK-8312492: Remove THP sanity checks at VM startup

### PR DESCRIPTION
**Pull request by tstuefe**
When starting the JVM with -XX:+UseTransparentHugePages, we need to know if the OS supports THPs:

A) We could run on a super-old kernel (very unlikely, THPs are supported since 2.6.38 which is EOL since 2011)
B) We could run on a kernel built without THP support, e.g. an embedded device
C) We could run on a system with THPs disabled by the admin

The JVM does a little sanity test at startup: it maps an area the size of a huge page - but in small pages - then calls madvise(MADV_HUGEPAGE) on it. The JVM assumes if the madvise() worked, all is fine.

That sanity test may be expensive since we allocate (reserve AND commit) an area, then signal khugepaged to fold it into a huge page. That area could be large once the kernel supports 1 GB THP pages. We may end up committing 1 GB space at startup. In addition to that, depending on the THP defrag settings, the allocation may stall. Even if it does not stall it creates unnecessary work for both the JVM and khugepaged.

The sanity test is also not really useful since it will happily report success on systems with THPs disabled (case (C)). Arguably, that is the majority of cases on existing Linux installations.

The correct way is to check the proc fs whether THPs are enabled. But we already do that since [JDK-8310233](https://bugs.openjdk.org/browse/JDK-8310233): "Fix THP detection on Linux". And that test detects all cases (A)(B)(C). Therefore we can completely remove the sanity checks for THPs.

I tested the patch manually in a number of settings with THPs enabled, disabled, and not compiled into the kernel at all. There is no behavioral difference because since [JDK-8310233](https://bugs.openjdk.org/browse/JDK-8310233) we catch all relevant cases before ever entering this sanity test.

[1] https://github.com/openjdk/jdk/blob/37ca9024ef59d99cae0bd7e25b2e6d3c1e085f97/src/hotspot/os/linux/os_linux.cpp#L3762

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8312492](https://bugs.openjdk.org/browse/JDK-8312492): Remove THP sanity checks at VM startup (**Bug** - P4)

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Coleen Phillimore](https://openjdk.org/census#coleenp) (@coleenp - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14967/head:pull/14967` \
`$ git checkout pull/14967`

Update a local copy of the PR: \
`$ git checkout pull/14967` \
`$ git pull https://git.openjdk.org/jdk.git pull/14967/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14967`

View PR using the GUI difftool: \
`$ git pr show -t 14967`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14967.diff">https://git.openjdk.org/jdk/pull/14967.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14967#issuecomment-1645194880)

### CODE REVIEW

File: src/hotspot/os/linux/os_linux.cpp
Comment (tstuefe) [inline src/hotspot/os/linux/os_linux.cpp:2808]: Reviewer note: I remove this since I am sure nobody builds on Linux < 2.6.38 - and if we would, probably a lot of other stuff would break first. I am unemotional though and could leave it in if desired.

**Review by dholmes-ora — APPROVED** [review]
Seems quite reasonable.

Thanks.

**Comment by tstuefe** [comment]
> Seems quite reasonable.
> 
> Thanks.

Thank you David.

**Comment by tstuefe** [comment]
x86 error unrelated.

A second review would make me happy.

**Comment by tstuefe** [comment]
Friendly ping. Need a second review.

**Comment by tstuefe** [comment]
Thank you @coleenp!

/integrate

**Pull request closed without merging**
